### DISCUSSION TITLE
How to upgrade game to new version?

### DISCUSSION THREAD

**Discussion by alexit190** [lanedirt/OGameX#385] [General]
Hello Lanedirt, thank you for this project, I like ogame and ogamex looks cool.
I see there is new version 0.8, how do I upgrade my installation from version 0.7? I would like to continue my progess in ogame.
I used this way to install game: Production: Install OGameX using Docker (Windows)

**Comment by lanedirt** [comment]
Hi @alexit190,

Nice to hear that you're using OGameX 😄!

Good point regarding upgrading, I see that there is no documentation for that yet. I'll try adding it to the readme so it's more clear for everyone.

Right now from the top of my head the following instructions should work to upgrade the `Production` install. Can you test if this works?

---

1. Open a terminal and navigate to the root directory of where you previously cloned the repository. By default this folder is called `OGameX` and contains the `docker-compose.yml` file. 

2. Fetch new Git tags and checkout the new stable version.
  ```
  $ git fetch --tags
  $ git checkout 0.8.0
  ```

3. Rebuild the docker containers:
  ```
  $ docker compose -f docker-compose.prod.yml up -d --build --force-recreate
  ```

4. Access the "ogame-app" Docker container:
  ```
  $ docker exec -it ogame-app /bin/bash
  ```

5. Update composer dependencies, run Laravel migrations and clear the caches.
  ```
  $ composer install --no-dev
  $ php artisan migrate --force
  $ php artisan cache:clear && php artisan config:cache && php artisan route:cache && php artisan view:cache
  ```

Now your game should be updated to the latest version with all data preserved.

**Comment by TheChaser5** [comment]
Hello lanedirt,

Can you give me the exact commands for this:

$ git fetch --tags
$ git checkout 0.8.0

I'm not that sharp with linus commands.

Thank you in advance!

Edit: I found it, you need to do the commands inside the folder ^^ thank you
